cores/esp32/esp32-hal-misc.c

```diff
@@ -142,6 +144,7 @@ unsigned long IRAM_ATTR millis()
 void delay(uint32_t ms)
 {
     vTaskDelay(ms / portTICK_PERIOD_MS);
+    if (yield != __yield) yield();
```

Comment:
uhmmm ... no! no changes to delay ;) in FreeRTOS delay does yield.

Reply:
@me-no-dev I am not even sure that using FreeRTOS tasks, that is, starting AND destroying them, is such a great idea with ESP32 Arduino, as my reading of the docs is that it takes the IDLE task to clean up destroyed tasks completely. Generally I don't see Arduino sketches letting the IDLE task run at all, unless *Yield defers to lower priority tasks?! But no one is required to use *Delay in a sketch, right?

Haven't I proved beyond reasonable doubt that AVR Arduino, as the gold standard, calls the OVERRIDABLE Arduino yield() as part of any Arduino delay()? Everyone is free to use FreeRTOS calls on ESP32, but Arduino libs and Arduino sketches don't and so they can rightfully expect Arduino compatibility, don't you think so, too?

I've learned from discussions here that there is such a thing as using Arduino-esp32 as a so called "component". There's already an incompatibility with that mode in the much lower tick frequency. So maybe, could your propose some #ifdef define for me to use around the above code change, such it will be disabled in the component mode?

I would also be quite willing to research into an even more transparent way of shimming vTaskDelay(), the issue being how to learn the xTicksToDelay value. I haven't found any.